disnake/abc.py

```diff
@@ -974,10 +974,11 @@ async def _clone_impl(
         base_attrs: Dict[str, Any],
         *,
         name: Optional[str] = None,
+        category_id: Optional[Snowflake] = None,
         reason: Optional[str] = None,
     ) -> Self:
         base_attrs["permission_overwrites"] = [x._asdict() for x in self._overwrites]
-        base_attrs["parent_id"] = self.category_id
+        base_attrs["parent_id"] = category_id or self.category_id
```

Comment:
`category_id` is `Optional[Snowflake]` while `self.category_id` is `Optional[int]`